src/plugins/borrow-plugins/types.ts

```diff
@@ -101,6 +109,7 @@ export type BorrowEngine = {
 
   // Close loan account
   close: () => Promise<ApprovableAction>
+  repayWithCollateral: (request: RepayWithCollateralRequest) => Promise<ApprovableAction>
```

Comment:
I actually meant to remove close().

We can't iteratively close because we can't setup multiple swapAndRepay calls without the risk of volatility possibly causing queued calls to be rejected. Also, per my discussion with @paullinator, one of the requirements for close is to end up with a single BTC asset in the user's wallet without incurring additional tax implications with swaps.